docs/dep/dep-0001.rst

```diff
@@ -0,0 +1,599 @@
+circular.).
+
+Once a DEP has been accepted, the reference implementation must be
+completed. When the reference implementation is complete and incorporated
+into the main source code repository, the status will be changed to "Final".
+
+A DEP can also be assigned status "Deferred". The DEP author or an
+editor can assign the DEP this status when no progress is being made
+on the DEP. Once a DEP is deferred, a DEP editor can re-assign it
+to draft status.
+
+A DEP can also be "Rejected". Perhaps after all is said and done it
+was not a good idea. It is still important to have a record of this
+fact. The "Withdrawn" status is similar - it means that the DEP author
+themselves has decided that the DEP is actually a bad idea, or has
+accepted that a competing proposal is a better alternative.
+
+When a DEP is Accepted, Rejected or Withdrawn, the DEP should be updated
+accordingly. In addition to updating the status field, at the very least
+the Resolution header should be added with a link to the relevant post
+in the debops-users mailing list archives.
+
+DEPs can also be superseded by a different DEP, rendering the original
+obsolete. This is intended for Informational DEPs, where version 2 of
+an API can replace version 1.
+
+The possible paths of the status of DEPs are as follows:
+
+.. image:: dep-0001-1.png
+
+Some Informational and Process DEPs may also have a status of "Active"
+if they are never meant to be completed. E.g. DEP 1 (this DEP).
+
+
+DEP Maintenance
+~~~~~~~~~~~~~~~
+
+In general, Standards track DEPs are no longer modified after they have reached
+the Final state. Once a DEP has been completed, the Language and standard
+environment references become the formal documentation of the expected
+behavior.
+
+Informational and Process DEPs may be updated over time to reflect changes
+to development practices and other details. The precise process followed in
+these cases will depend on the nature and purpose of the DEP being updated.
+
+
+What belongs in a successful DEP?
+---------------------------------
+
+Each DEP should have the following parts:
+
+1. Preamble -- reStructuredText field list style headers containing
+   meta-data about the DEP, including the DEP number, a short
+   descriptive title (limited to a maximum of 44 characters), the names,
+   and optionally the contact info for each author, etc.
+
+2. Abstract -- a short (~200 word) description of the technical issue
+   being addressed.
+
+3. Copyright/public domain -- Each DEP must either be explicitly labeled as
+   placed in the public domain or licensed under the
+   `GNU General Public License v3`_.
+
+4. Specification -- The technical specification should describe the
+   syntax and semantics of any new project feature. The importance of various
+   specification details can be emphasised by usage of the capitalized key
+   words, as described in the `RFC 2119`__.
+
+   .. __: https://tools.ietf.org/html/rfc2119
```

Comment:
"must" (lower case) is used a lot in the document and the meaning is defined by RFC 2119. Although the RFC seems to be not 100 % precise about upper/lower case. I would say it is better to always write them in upper case (as done in the keyring docs) so that it is clear that the author was aware of the meaning. That is also what the RFC implies and what it uses in itself.

Also because you write "details can be emphasised", if we don’t use upper case then, it is unclear.

Reply:
[RFC 8174](https://tools.ietf.org/html/rfc8174) makes this more precise - only UPPERCASE words have the intended meaning defined by the RFC 2119.

In RFC 2119 there's a section about usage of the terms which states:

> Imperatives of the type defined in this memo must be used with care
> and sparingly.  In particular, they MUST only be used where it is
> actually required for interoperation or to limit behavior which has
> potential for causing harm

In my understanding it means that if you are writing a specification for an obvious thing that cannot be interpreted in multiple ways, using lowercase version is sufficient, as in speaking in normal voice. Only ambigious things require an uppercase version of the words which then carry **very specific meaning** defined by the RFC and that causes the given specification to be interpreted in a certain way, intended by the author.

Reply:
Works for me. To be explicit, we should the use:


```
      The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
      NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED",
      "MAY", and "OPTIONAL" in this document are to be interpreted as
      described in BCP 14 [RFC2119] [RFC8174] when, and only when, they
      appear in all capitals, as shown here.
```